File: src/dev/danzou/scala/Day1.scala
package dev.danzou.scala

import dev.danzou.utils.Utils.readInputString

object Day1 {
  def main(args: Array[String]): Unit = {
    val input = readInputString()

    val sum = input
      .split("\n\n")
      .map({ _.split("\n").map({ _.toLong }) })
      .map({ _.sum })
      .sorted
      .takeRight(3)
      .sum

    println(sum)

    assert(209481 == sum)
  }
}

File: src/dev/danzou/kotlin/day1.kt
package dev.danzou.kotlin

import dev.danzou.utils.Utils.readInputString
import kotlin.test.assertEquals

fun main() {
    val input = readInputString()

    val sums = input
        .split("\n\n")
        .map { it.split("\n").map { it.toLong() } }
        .map { it.sum() }

    val part1 = sums.max()

    println("Part 1: $part1")

    val part2 = sums
        .sorted()
        .takeLast(3)
        .sum()

    println("Part 2: $part2")
    assertEquals(209481, part2) // Part 2
}